libyul/AsmParser.cpp

```diff
@@ -78,14 +81,70 @@ unique_ptr<Block> Parser::parse(std::shared_ptr<Scanner> const& _scanner, bool _
 	return nullptr;
 }
 
+langutil::Token Parser::advance()
+{
+	auto const token = ParserBase::advance();
+	if (m_sourceMap)
+		fetchSourceLocationFromComment();
+	return token;
+}
+
+void Parser::fetchSourceLocationFromComment()
+{
+	solAssert(m_sourceMap.has_value(), "");
+
+	if (m_scanner->currentCommentLiteral().empty())
+		return;
+
+	static std::regex const lineRE = std::regex(
+		R"~~~(\s*@src\s+(\d+):(-1|\d+):(-1|\d+)\s*)~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+
+	string_view text = m_scanner->currentCommentLiteral();
+	std::cmatch cm;
+	if (!std::regex_search(text.data(), text.data() + text.size(), cm, lineRE))
+		return;
+
+	try
+	{
+		solAssert(cm.size() == 4, "");
+		auto const sourceIndex = stoul(cm[1].str());
+		solAssert(sourceIndex <= std::numeric_limits<unsigned long>::max(), "");
+		auto const start = stoi(cm[2].str());
+		auto const end = stoi(cm[3].str());
+		solAssert(start <= end, "");
```

Comment:
I don't think we can assert that.

Reply:
so add an error report for that, too. okay.